src/recon.rs

```diff
@@ -4525,22 +4515,22 @@ pub(crate) unsafe fn rav1d_backup_ipred_edge<BD: BitDepth>(
             + (((t.b.y + f.sb_step) * 4 >> ss_ver) - 1) as isize * BD::pxstride(f.cur.stride[1]);
         let mut pl = 1;
         while pl <= 2 {
+            let ipred_edge_off =
+                (f.ipred_edge_off * pl) + (sby_off + (x_off * 4 >> ss_hor)) as usize;
+            let n = (4 * (ts.tiling.col_end - x_off) >> ss_hor)
+                .try_into()
+                .unwrap();
             BD::pixel_copy(
-                &mut slice::from_raw_parts_mut(
-                    f.ipred_edge[pl as usize].cast(),
-                    (sby_off + (x_off * 4 >> ss_hor) + (4 * (ts.tiling.col_end - x_off) >> ss_hor))
-                        .try_into()
-                        .unwrap(),
-                )[(sby_off + (x_off * 4 >> ss_hor)).try_into().unwrap()..],
+                &mut f
+                    .ipred_edge
+                    .mut_slice_as(ipred_edge_off..ipred_edge_off + n),
```

Comment:
```suggestion
                    .mut_slice_as((ipred_edge_off.., ..ipred_edge_off)),
```

Reply:
Doesn't work with the `slice_as` methods.